File: Lab3ChausKI306/src/KI/Chaus/Lab3/MilitaryAlpinistEquipment.java
package KI.Chaus.Lab3;

/**
 * This class represents military alpinist equipment and extends the AlpinistEquipment class.
 * It provides mechanisms for correct functioning specific to military equipment.
 * It also implements the MilitaryEquipment interface.
 * 
 * @author Chaus Bohdan
 * @version 1.0
 */
public class MilitaryAlpinistEquipment extends AlpinistEquipment implements MilitaryAlpinistEquipmentInterface {

    // Indicates whether night vision mode is enabled.
    private boolean nightVisionEnabled;

    // Constructor initializes nightVisionEnabled to false by default.
    public MilitaryAlpinistEquipment() {
        nightVisionEnabled = false;
    }

    /**
     * Enables night vision mode.
     */
    public void enableNightVision() {
        nightVisionEnabled = true;
    }

    /**
     * Disables night vision mode.
     */
    public void disableNightVision() {
        nightVisionEnabled = false;
    }

    /**
     * Checks if night vision mode is enabled.
     * 
     * @return true if night vision is enabled, false otherwise.
     */
    public boolean isNightVisionEnabled() {
        return nightVisionEnabled;
    }

    /**
     * Overrides the method in the interface.
     * Prints a message indicating a military alpinist action is being performed.
     */
    @Override
    public void performMilitaryAction() {
        System.out.println("Performing military alpinist action...");
    }

    /**
     * Overrides the method in the interface.
     * Indicates if the equipment is combat-ready based on night vision status.
     * 
     * @return true if night vision is enabled, false otherwise.
     */
    @Override
    public boolean isCombatReady() {
        return isNightVisionEnabled();
    }
}

File: Lab9ChausKI306/MilitaryAlpinistEquipment.py
from AlpinistEquipment import AlpinistEquipment

class MilitaryAlpinistEquipment(AlpinistEquipment):
    def __init__(self):
        # Call the constructor of the parent class AlpinistEquipment
        super().__init__()
        
        # Initialize an additional attribute specific to MilitaryAlpinistEquipment
        self.night_vision_enabled = False

    def enable_night_vision(self):
        # Enable night vision
        self.night_vision_enabled = True

    def disable_night_vision(self):
        # Disable night vision
        self.night_vision_enabled = False

    def is_night_vision_enabled(self):
        # Check if night vision is enabled
        return self.night_vision_enabled

    def perform_military_action(self):
        # Perform a military alpinist action
        print("Performing military alpinist action...")

    def is_combat_ready(self):
        # Check if the equipment is combat ready, based on night vision availability
        return self.is_night_vision_enabled()
